### DISCUSSION TITLE
Flow reversal problem

### DISCUSSION THREAD

**Discussion by abhi1587** [idaholab/moose#26977] [Q&A Modules: Thermal Hydraulics] [answered]
Hi, I am trying to simulate a conjugate heat transfer problem with the time-dependent mass flow rate in a pipe with m1, m2, and m3 mass flow rates in different periods of simulation. 
m1 and m3 being finite positive values and m2 is 0. Attached is the script. 
The issue is when the mass flow is reduced to 0, the center line velocity reduces to a negative value and so does the temperature. The figures are attached. 

It would be helpful if anyone could take a look at my script and point out the issue in it or suggest some checks to perform.

Thanks

[Simulation_main.txt](https://github.com/idaholab/moose/files/14498716/Simulation_main.txt)
[Mesh.txt](https://github.com/idaholab/moose/files/14498740/Mesh.txt)

<img width="761" alt="Temperature" src="https://github.com/idaholab/moose/assets/8911653/5b759ea0-df5c-4084-b432-53f1b281cf0f">
<img width="766" alt="Velocity_Z" src="https://github.com/idaholab/moose/assets/8911653/2ed88bd6-b9fd-4819-a273-e4221531f22a">

**Comment by hugary1995** [comment]
@joshuahansel Could you help us with this test setup when you get a chance? @abhi1587 is seeing a decrease in solid temperature even when the fluid flow is turned off, and we suspect it's coming from the negative velocity. Thanks.

**Comment by joshuahansel** [comment, marked as answer]
I'm sorry, I started to look at your input and then ran into some VSCode issues and moved to something else while that was getting worked out.

Could you try a larger number of axial elements than 3? 100 would be nice. I realize you may have convergence issues because of your 3D heat structure coupling not having full Jacobians. While troubleshooting, can you switch to a 2D cylindrical heat structure. That'll converge nicely. Here's an example:
```
moose/modules/thermal_hydraulics/test/tests/components/heat_transfer_heat_structure_1phase/phy.T_wall_transfer_3eqn_x.i
```
I suspect this is maybe caused by boundary conditions too close to each other and fighting, plus maybe the equation of state you're using, which I'm not super familiar with. Second step would be to change that to something else.

**Reply by abhi1587** [reply]
Thanks, Joshua. Changing the equation of state solved the issue. I will look into the documentation for how are these models affecting the solution.
